pkg/utils/pod/scheduling.go

```diff
@@ -207,3 +207,8 @@ func HasPodAntiAffinity(pod *corev1.Pod) bool {
 		(len(pod.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution) != 0 ||
 			len(pod.Spec.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution) != 0)
 }
+
+// HasDRARequirements returns true if ResourceClaims are defined in the pod spec
+func HasDRARequirements(pod *corev1.Pod) bool {
```

Comment:
This is probably fine in practice, but technically we shouldn't fail scheduling just because `ResourceClaims` are defined for the pod. Really we only care if any of the containers consume them. For correctness sake, I'd check if any containers have requests for claims, rather than just checking if claims exist.